src/AppInstallerCommonCore/Manifest/ManifestSchemaValidation.cpp

```diff
@@ -0,0 +1,207 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+#include "pch.h"
+#include "winget/Yaml.h"
+#include "winget/ManifestCommon.h"
+#include "winget/ManifestSchemaValidation.h"
+#include "winget/ManifestYamlParser.h"
+
+#include <ManifestSchema.h>
+
+namespace AppInstaller::Manifest::YamlParser
+{
+    using namespace std::string_view_literals;
+
+    namespace
+    {
+        enum class YamlScalarType
+        {
+            String,
+            Int
+        };
+
+        // List of fields that use non string scalar types
+        const std::map<std::string_view, YamlScalarType> ManifestFieldTypes=
+        {
+            { "InstallerSuccessCodes"sv, YamlScalarType::Int }
+        };
+
+        YamlScalarType GetManifestScalarValueType(const std::string& key)
+        {
+            auto iter = ManifestFieldTypes.find(key);
+            if (iter != ManifestFieldTypes.end())
+            {
+                return iter->second;
+            }
+
+            return YamlScalarType::String;
+        }
+
+        Json::Value YamlScalarNodeToJson(const YAML::Node& scalarNode, YamlScalarType scalarType)
+        {
+            if (scalarType == YamlScalarType::Int)
+            {
+                return Json::Value(scalarNode.as<int>());
+            }
+            else
+            {
+                return Json::Value(scalarNode.as<std::string>());
+            }
+        }
+
+        Json::Value ManifestYamlNodeToJson(const YAML::Node& rootNode, YamlScalarType scalarType = YamlScalarType::String)
+        {
+            Json::Value result;
+
+            if (rootNode.IsNull())
+            {
+                result = Json::Value::nullSingleton();
+            }
+            else if (rootNode.IsMap())
+            {
+                for (auto const& keyValuePair : rootNode.Mapping())
+                {
+                    // We only support string type as key in our manifest
+                    auto key = keyValuePair.first.as<std::string>();
+                    result[keyValuePair.first.as<std::string>()] = ManifestYamlNodeToJson(keyValuePair.second, GetManifestScalarValueType(key));
+                }
+            }
+            else if (rootNode.IsSequence())
+            {
+                for (auto const& value : rootNode.Sequence())
+                {
+                    result.append(ManifestYamlNodeToJson(value, scalarType));
+                }
+            }
+            else if (rootNode.IsScalar())
+            {
+                result = YamlScalarNodeToJson(rootNode, scalarType);
+            }
+            else
+            {
+                THROW_HR(E_UNEXPECTED);
+            }
+
+            return result;
+        }
+    }
+
+    std::string LoadResourceAsString(PCWSTR resourceModuleName, PCWSTR resourceName, PCWSTR resourceType)
+    {
+        HMODULE resourceModule = GetModuleHandle(resourceModuleName);

```

Comment:
>GetModuleHandle [](start = 33, length = 15)

If you are going to allow arbitrary modules here, you should use the proper, thread safe GetModuleHandleEx and a wil::unique_hmodule to hold it.